pkg/capacity/capacity.go

```diff
@@ -99,39 +102,61 @@ type FlavorInfo struct {
 	Labels     map[string]string
 }
 
-func NewCapacity(cap *kueue.Capacity) *Capacity {
+func NewCapacity(cap *kueue.Capacity) (*Capacity, error) {
+	usedResources := make(Resources, len(cap.Spec.RequestableResources))
+	for _, r := range cap.Spec.RequestableResources {
+		if len(r.Flavors) == 0 {
+			continue
+		}
+
+		ts := make(map[string]int64, len(r.Flavors))
+		for _, t := range r.Flavors {
+			ts[t.Name] = 0
+		}
+		usedResources[r.Name] = ts
+	}
+
 	c := &Capacity{
-		Name:                 cap.Name,
-		RequestableResources: resourcesByName(cap.Spec.RequestableResources),
-		UsedResources:        make(Resources, len(cap.Spec.RequestableResources)),
-		Workloads:            map[string]*workload.Info{},
-		QueueingStrategy:     cap.Spec.QueueingStrategy,
+		Name:             cap.Name,
+		UsedResources:    usedResources,
+		Workloads:        map[string]*workload.Info{},
+		QueueingStrategy: cap.Spec.QueueingStrategy,
+	}
+	if err := c.update(cap); err != nil {
+		return nil, err
 	}
 
+	return c, nil
+}
+
+func (c *Capacity) update(in *kueue.Capacity) error {
+	c.RequestableResources = resourcesByName(in.Spec.RequestableResources)
+	nsSelector, err := metav1.LabelSelectorAsSelector(in.Spec.NamespaceSelector)
+	if err != nil {
+		return err
+	}
+	c.NamespaceSelector = nsSelector
 	labelKeys := map[corev1.ResourceName]sets.String{}
-	for _, r := range cap.Spec.RequestableResources {
+	for _, r := range in.Spec.RequestableResources {
```

Comment:
Is it possible that the new flavor is added? Then we need to add this items in the  `c.UsedResources` ?
Because it's also invoked in `UpdateCapacity`.

Reply:
good catch, updated